docs/LoadAlgorithm.md

```diff
@@ -0,0 +1,164 @@
+     - Disables transaction logging.
+     - Disables transactions on storage  backend.
+     - Bigger dirty transaction cache.
+     - Disables external vertex existence checks.
+     - Disables consistency checks (verify uniqueness, acquire locks etc)
+
+3. ```cache.db-cache```: Better to turn this off, since cache is not distributed. Therefore each node will have a different view of the data.
+4. ```threadPoolWorker```: The default value for this is 1. This is essentially a traffic cop to disseminate work. It should be set at most 2 * number of cores in the JanusGraph instance.
+5. ```gremlinPool```: The number of traversals that can be run in parallel. This pool runs until query completes. 
+
+
+### 2. Load schema
+This will involve creating
+  - Vertex Labels
+    Each vertex will have one label which is a kind of its 'type'.
+  - Edge Labels (Still to describe what will be edge label)
+  - Property Keys
+    - Vertex property ```Rowkey```
+    - All [```link```](Schema-Model.md#link-properties) properties as vertex properties.
+
+### 3. Read Parquet Data
+
+- Spark job will read the partitioned data directory for the data to be ingested for the particular day.
+- This means we should invoke the job with ```startDate``` param.
+- We should also control overwrite capability in case this data is already ingested?
+
+### 4. Load Vertices and Edges
+
+Every alert (single row in hbase) will be modelled as a graph vertex.
+
+<div id="option1" />
+#### Option 1
+
+This approach focusses on using bulk loading offered by JanusGraph to push the data.
+
+- Read all the rows, group by the interesting columnnames (link properties).
+- Vertices in each group become ```cliques``` of the graph, since each of them needs to be connected to all the others in the group.
+- Load all these as vertices of the graph and add the label and properties.
+- Then for each of these groups:
+   - **Search for all vertices with the same link properties.**
+   - For each vertex in the group, add edges to all the returned vertices in the above step.
+
+##### Advantages
+
+- Fully reliant on Janusgraph
+
+##### Disadvantages
+
+- A LOT of read queries required. This will definitely slow down the process.
+- As data grows, time taken by the queries will increase, and so will the overall data loading time.
+
+#### Option 2
+
+This approach involves, directly generating HBase Files from spark job, in a format that
+JanusGraph would write, if we used [Option1](#option1)
+
+- There is already an [open ticket in JanusGraph](https://github.com/JanusGraph/janusgraph/issues/885)
+- This uses the bulk loading mechanism in Hbase to achieve high write speeds
+- Example of [bulk-loading hbase using spark](https://www.opencore.com/blog/2016/10/efficient-bulk-load-of-hbase-using-spark/)
+- **But this approach cannot be used because we want to add edges to existing vertices while writing data**
+  This would involve updating existing data files, by reading, modifying and rewriting them back.
```

Comment:
This sounds a no-go, as we will have to add edges to existing vertices while writing data each time. So the associated I/O cost would be overwhelming...